### ISSUE TITLE
Update system to work with new OpenMCT API

### ISSUE THREAD

**Issue by crgimenes** [crgimenes/otp#4]
We need to upgrade the system to work with the new OpenMCT API but NASA is still making changes to the system and there is virtually no documentation of the new API. This issue is just to keep this in mind and change the system as soon as the new API is stable.

**Comment by willmendil** [comment]
The new api is up and running with clear documentation, are you going to update your repo? if not i'll start from scratch. just tell me, we can even work at it together if you want

**Comment by crgimenes** [comment]
hello Will,

I want to update the project and I like the idea of working together.
I'll take a look at the new documentation and see what the project looks like today.

I am grateful for your help.
What do you think about opening some issues with the things we need to do and we follow from there?

**Comment by willmendil** [comment]
Great! Give me a bit of time to understand the api, and see how to build from it. I've just cloned your repo and made a new branch for me, i'll push it when I have a bit of progress. I'll post issues when I identify specific points. 

**tl;dr** yes, give me time to better understand the nasa api

**Comment by crgimenes** [comment]
Hi Will,

Some tips to make our work smoother.

Do not create a branch for yourself, instead create a branch for each feature you are going to develop and then create a pull request to be reviewed and the feature enters the master. This way we can isolate each feature and avoid collision of our work. After the PR review and merge we simply delete the branch and close the issue.

It is always good to discuss what you want to develop in an issue before writing code, so we guarantee that we agree with what is being done and we also generate some documentation if other people want to participate. Of course, you do not need to create an issue if it is something obvious, such as the correction you made, but it is always good to do a PR.

Always check the other developers to review the PR, so we make sure that everyone is in agreement before anything gets into the master.

Sorry if these things are obvious to you, I just wanted to make sure we're on the same page.

Thank you for your help!

**Comment by willmendil** [comment]
Cool tips, i have to be honnest, I have never worked on a project with other people, so your insight is quite valuable.  I'll do as you say. For the moment I'm readying the documentation in details in order to understand how making a golang interface would be done. I guess we can keep this issue for open discussion, and will make specific issue/branches for each feature. This is your project, so I'll follow your guide lines. Maybe later, creating a project could be an idea? what do you think?

**Comment by crgimenes** [comment]
Working on open source projects is very good, I learned a lot from it.

As for the API, I still haven't read the new documentation but I suspect that the operation must not have changed much. 

The way it worked before was the front end which is in _node_ consuming a websocket provided by the **telemetry provider** (our project). 

This websocket is very simple, basically it is a pub/sub interface with some specific calls. The initial part is the front-end asking for the taxonomy dictionary via websocket, with that, the front-end can assemble the menu, and through this menu the user can command the front-end to read/subscribe the sensors channel that he finds interesting. So, all via websocket, the front-end subscribes to a sensor channel and our code simply sends data through that channel, and do this for everyone who subscribes to the sensor channel.

The current code already does it all, and the user of our package can use our code to add sensors by simple hooking functions, and just worrying about reading the sensor, the distribution to the front end is up to us.

I believe that what must have changed is the structure of the taxonomy dictionary, keywords, etc., but the basic structure must be the same.

About creating other projects, I do this every time I have a new idea. For example, this repository is for the package, which is sure to be used by other projects. If one day this package grows enough we will transform it into an organization and create sub-projects that are useful under that organization, for now it is in my repos just because it is not big enough to justify having an organization. So this is no longer just my project, it is our project, and everyone is willing to contribute. This is free software after all.

**Comment by willmendil** [comment]
him i have to be honnest, stuff have cought me up after the covid so i won't have time to work in this project. I'll still follow your repo if you ever find time, but i won't be able to work on it. my apologies

**Comment by crgimenes** [comment]
Hello Will, I hope you are well. I'm also having a hard time putting my plans into action, little time and a lot to do.
I am trying to read the telemetry generated by my model aircraft and for the time being I am still working on the hadrware.
You are welcome to participate whenever you want.

**Comment by willmendil** [comment]
so am i, i'm working on a small tank, i want to try it with gobot bocv etc... so i'm just slowlly getting all the modules working for now and hope to have a relemetry at the end
